packages/trader/src/App/Components/Elements/VideoStream/video-stream.tsx

```diff
@@ -1,75 +1,299 @@
+        if (!video_ref?.current) return;
+
+        video_ref.current.muted = new_value;
+    };
+
+    const onPlaybackRateChange = (new_value: number) => {
+        if (!video_ref?.current) return;
+
+        video_ref.current.playbackRate = new_value;
+    };
+
+    const togglePlaying = (e: React.MouseEvent<HTMLElement, MouseEvent> | React.KeyboardEvent<HTMLElement>) => {
+        e.stopPropagation();
+
+        if (!video_ref?.current || !progress_bar_ref?.current) return;
+
+        if (is_playing) {
+            video_ref.current.pause();
+            progress_bar_ref.current.style.setProperty('transition', 'none');
+            cancelAnimationFrame(animation_ref.current);
+        } else {
+            video_ref.current.play();
+        }
+
+        setIsPlaying(prev => !prev);
+        setIsEnded(false);
+    };
+
+    const calculateNewWidth = (e: React.MouseEvent<HTMLElement, MouseEvent> | React.TouchEvent<HTMLElement>) => {
+        const progress_bar = document.querySelector('.player__controls__progress-bar');
+        const client_X =
+            e.type === 'mousemove' || e.type === 'click'
+                ? (e as React.MouseEvent<HTMLElement, MouseEvent>).clientX
+                : (e as React.TouchEvent<HTMLElement>).changedTouches[0].clientX;
+
+        let new_width =
+            ((client_X - shift_X - (progress_bar?.getBoundingClientRect().left ?? 0)) /
+                (progress_bar?.getBoundingClientRect().width ?? 0)) *
+            100;
+        if (new_width >= 100) {
+            new_width = 100;
+        }
+        if (new_width <= 0) {
+            new_width = 0;
+        }
+
+        return new_width;
+    };
+
+    const dragStartHandler = (e: React.MouseEvent<HTMLElement, MouseEvent> | React.TouchEvent<HTMLElement>) => {
+        if (e.type === 'mousedown') e.preventDefault();
+        e.stopPropagation();
+
+        const point = document.querySelector('.player__progress-dot') as HTMLElement;
+        const client_X =
+            e.type === 'mousedown'
+                ? (e as React.MouseEvent<HTMLElement, MouseEvent>).clientX
+                : (e as React.TouchEvent<HTMLElement>).changedTouches[0].clientX;
+        setShiftX(client_X - (point?.getBoundingClientRect().left ?? 0));
+
+        video_ref?.current?.pause();
+        setIsPlaying(false);
+
+        is_dragging.current = true;
+
+        if (is_mobile) {
+            point.style.width = '1.6rem';
+            point.style.height = '1.6rem';
+            point.style.bottom = '-0.6rem';
+        }
+    };
+
+    const dragMoveHandler = (e: React.MouseEvent<HTMLElement, MouseEvent> | React.TouchEvent<HTMLElement>) => {
+        if (e.type === 'mousemove') e.preventDefault();
+        e.stopPropagation();
+
+        if (!is_dragging.current) return;
+        if (!video_ref?.current || !progress_bar_ref?.current) return;
+
+        const new_width = calculateNewWidth(e);
+        progress_bar_ref.current.style.setProperty('transition', 'none');
```

Comment:
We are setting `setProperty('transition', 'none')` to `progress_bar_ref` element for 3 times and to `volume_bar_ref` for `2 times`.
**Let's move all the style settings** **to CSS** by adding classNames to these elements based on some state. This includes all non-calculated ones, **not only in this file**:
```
style.setProperty('transition', 'none')
style.setProperty('transition', 'all 0.3s linear')
style.setProperty('height', '0%')
style.width
style.height
style.bottom
```
And in our callbacks, we would only need to change the react state which would result in the correct className being assigned to each element.